apps/server/src/modules/h5p-editor/uc/h5p.uc.ts

```diff
@@ -0,0 +1,384 @@
+				query.minorVersion,
+				language,
+				user
+			);
+
+			return result;
+		} catch (err) {
+			throw this.mapH5pError(err);
+		}
+	}
+
+	public async postAjax(
+		currentUser: ICurrentUser,
+		query: AjaxPostQueryParams,
+		body: AjaxPostBodyParams,
+		contentFile?: Express.Multer.File,
+		h5pFile?: Express.Multer.File
+	) {
+		const user = this.changeUserType(currentUser);
+		const language = await this.getUserLanguage(currentUser);
+
+		try {
+			const result = await this.h5pAjaxEndpoint.postAjax(
+				query.action,
+				body,
+				language,
+				user,
+				contentFile && {
+					data: contentFile.buffer,
+					mimetype: contentFile.mimetype,
+					name: contentFile.originalname,
+					size: contentFile.size,
+				},
+				query.id,
+				undefined,
+				h5pFile && {
+					data: h5pFile.buffer,
+					mimetype: h5pFile.mimetype,
+					name: h5pFile.originalname,
+					size: h5pFile.size,
+				},
+				undefined // TODO: HubID?
+			);
+
+			return result;
+		} catch (err) {
+			throw this.mapH5pError(err);
+		}
+	}
+
+	public async getContentParameters(contentId: string, currentUser: ICurrentUser) {
+		const { parentType, parentId } = await this.h5pContentRepo.findById(contentId);
+		await this.checkContentPermission(currentUser.userId, parentType, parentId, AuthorizationContextBuilder.read([]));
+
+		const user = this.changeUserType(currentUser);
+
+		try {
+			const result = await this.h5pAjaxEndpoint.getContentParameters(contentId, user);
+
+			return result;
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getContentFile(
+		contentId: string,
+		file: string,
+		req: Request,
+		currentUser: ICurrentUser
+	): Promise<{
+		data: Readable;
+		contentType: string;
+		contentLength: number;
+		contentRange?: { start: number; end: number };
+	}> {
+		const { parentType, parentId } = await this.h5pContentRepo.findById(contentId);
+		await this.checkContentPermission(currentUser.userId, parentType, parentId, AuthorizationContextBuilder.read([]));
+
+		const user = this.changeUserType(currentUser);
+
+		try {
+			const rangeCallback = this.getRange(req);
+			const { mimetype, range, stats, stream } = await this.h5pAjaxEndpoint.getContentFile(
+				contentId,
+				file,
+				user,
+				rangeCallback
+			);
+
+			return {
+				data: stream,
+				contentType: mimetype,
+				contentLength: stats.size,
+				contentRange: range, // Range can be undefined, typings from @lumieducation/h5p-server are wrong
+			};
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getLibraryFile(ubername: string, file: string) {
+		try {
+			const { mimetype, size, stream } = await this.libraryService.getLibraryFile(ubername, file);
+
+			return {
+				data: stream,
+				contentType: mimetype,
+				contentLength: size,
+			};
+		} catch (err) {
+			throw new NotFoundException();
+		}
+	}
+
+	public async getTemporaryFile(
+		file: string,
+		req: Request,
+		currentUser: ICurrentUser
+	): Promise<{
```

Comment:
Doesn't match the FileDto

Reply:
Dann bitte ein neuen typ definieren.

Reply:
unresolving this as appearentlz nothing has been done?
The point is, the return type of a usecase deserves a name, to tell us what this is. Especially since appearently multiply usecases return an object of the same type.

Please define it as some kind of dto with a useful name, instead of an inline object definition.

Reply:
resolved: added GetLibraryFile